### PR TITLE
build on both 7.10 and 7.8 (WIP)

### PR DESCRIPTION
**Pull request by tranma**

### CODE REVIEW

**Comment by amosr** [comment]
groovy

**Comment by amosr** [comment]
:watch:

**Comment by jystic** [comment]
Superseded by https://github.com/ambiata/icicle/pull/320

File: src/Icicle/Source/Transform/Desugar.hs [hunk 1]
```diff
@@ -322,7 +323,10 @@ data Tree a n x
          [(Constructor, Tree a n x)] (Tree a n x)
  deriving (Functor, Foldable, Traversable, Show)
 
-
+instance Applicative (Tree a n) where
+  pure = return
```
Comment (jystic) [inline src/Icicle/Source/Transform/Desugar.hs]: fwiw, this doesn't work according to the docs, not sure what the effect is, but they say you're supposed to move the `return` implementation here and make `return = pure`

**Pull request closed without merging**